pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/src/main/java/org/apache/pinot/plugin/ingestion/batch/common/SegmentGenerationTaskRunner.java

```diff
@@ -71,6 +72,7 @@ public class SegmentGenerationTaskRunner implements Serializable {
   @Deprecated
   public static final String DEPRECATED_USE_LOCAL_DIRECTORY_SEQUENCE_ID = "local.directory.sequence.id";
   public static final String USE_GLOBAL_DIRECTORY_SEQUENCE_ID = "use.global.directory.sequence.id";
+  public static final String USE_UNIQUE_SEGMENT_NAME_POSTFIX = "use_unique.segment.name.postfix";
```

Comment:
maybe call this `append.uuid.to.segment.name.suffix`?

Reply:
Yes, it is not good but I want to use something which doesn't tie the property name to its implementation (which is why is didn't use `.uuid`). I was thinking of modifying the existing `segment.name.postfix` to handle this with templating instead of a new prop

Reply:
I'd suggest not creating dependency between this new config and `segment.name.postfix`. Currently UUID won't be appended if name postfix is provided, which can cause confusion.
Suggest making it `append.uuid.to.segment.name` and when it is configured, append an uuid regardless of whether postfix is configured. UUID is a common concept so it should be okay to include it in the config key.

Reply:
Cool. refactored now. But the segmentNameGenerator interface is ugly. The constructors should not be accepting arguments but just a properties map/class.
I have created an issue for it which I will take up later

https://github.com/apache/pinot/issues/8799